sync/clients/sync.py

```diff
@@ -109,7 +110,16 @@ def delete_project(self, project_id: str) -> dict:
         return self._send(self._client.build_request("DELETE", f"/v1/projects/{project_id}"))
 
     def _send(self, request: httpx.Request) -> dict:
-        response = self._client.send(request)
+        try:
+            for attempt in Retrying(stop=stop_after_attempt(3), wait=wait_fixed(5)):
```

Comment:
Adding some retry logic for `503`s so that API calls to e.g. dev/stage don't fail if our instances are sleeping.

Reply:
This aspect of our dev environment is probably better handled with DNS configuration to park the request until the server's ready per the documentation, but retrying is good to have in general. Is this a good time to all the clients here? Additionally, expand the scope to include other status codes: {408, 425, 429, 500, 502, 503, 504} and an incrementing wait to recover from tiny blips faster.

Reply:
yeah, this was a little step towards adding some more general retrying for resiliency purposes. I agree with ya about there probably being a better way to solve this for dev/stage. I'd be happy to add additional status codes to this diff while we're at it